src/main/java/org/commonjava/indy/service/httprox/config/ServiceProxyConfig.java

```diff
@@ -0,0 +1,190 @@
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.commonjava.indy.service.httprox.client.content.ContentRetrievalService.EVENT_PROXY_CONFIG_CHANGE;
+
+@Startup
+@ApplicationScoped
+public class ServiceProxyConfig
+{
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    public static final String USER_DIR = System.getProperty( "user.dir" ); // where the JVM was invoked
+
+    @Inject
+    transient EventBus bus;
+
+    @JsonProperty( "read-timeout" )
+    private String readTimeout;
+
+    public String getReadTimeout()
+    {
+        return readTimeout;
+    }
+
+    private volatile Retry retry;
+
+    private Set<ServiceConfig> services = Collections.synchronizedSet( new HashSet<>() );
+
+    public Set<ServiceConfig> getServices()
+    {
+        return services;
+    }
+
+    public Retry getRetry()
+    {
+        return retry;
+    }
+
+    @Override
+    public String toString()
+    {
+        return "ProxyConfiguration{" + "readTimeout='" + readTimeout + '\'' + ", retry=" + retry + ", services="
+                + services + '}';
+    }
+
+    @PostConstruct
+    void init()
+    {
+        load( true );
+        logger.info( "Proxy config, {}", this );
+    }
+
+    private static final String PROXY_YAML = "application.yaml";
+
+    /**
+     * Load proxy config from '${user.dir}/config/proxy.yaml'. If not found, load from default classpath resource.
+     */
+    public void load( boolean init )
+    {
+        File file = new File( USER_DIR, "config/" + PROXY_YAML );
+        if ( file.exists() )
+        {
+            logger.info( "Load proxy config from file, {}", file );
+            try(FileInputStream fis = new FileInputStream( file ))
+            {
+                doLoad( fis );
+            }
+            catch ( IOException e )
+            {
+                logger.error( "Load failed", e );
+                return;
+            }
+        }
+        else if ( init )
+        {
+            logger.info( "Load proxy config from classpath resource, {}", PROXY_YAML );
+            InputStream res = this.getClass().getClassLoader().getResourceAsStream( PROXY_YAML );
+            if ( res != null )
+            {
+                doLoad( res );
+            }
+        }
+        else
+        {
+            logger.info( "Skip loading proxy config - no such file: {}", file );
+        }
+    }
+
+    private transient String stateHash; // used to check whether the custom proxy.yaml has changed
+
+    private void doLoad( InputStream res )
+    {
+        try
+        {
+            String str = IOUtils.toString( res, UTF_8 );
+            String nextStateHash = DigestUtils.sha256Hex( str ).toUpperCase();
+            if ( nextStateHash.equals( stateHash ) )
+            {
+                logger.info( "Skip, NO_CHANGE" );
```

Comment:
Does the config change during the runtime? If not, we don't need to check it.

Reply:
I think it won't. Will remove this to make it simple. thanks.